pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,959 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use crate::pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    t3rn_primitives::reexport_currency_types!();
+    use codec::Encode;
+    use frame_support::{dispatch::DispatchResult, pallet_prelude::*, traits::Currency};
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::{
+        traits::{CheckedAdd, CheckedDiv, Saturating, Zero},
+        PerThing, Perbill, Percent,
+    };
+    use sp_std::{collections::btree_map::BTreeMap, convert::TryInto, prelude::*};
+    use t3rn_primitives::{
+        account_manager::{AccountManager, Settlement},
+        attesters::AttestersReadApi,
+        claimable::{BenefitSource, CircuitRole, ClaimableArtifacts},
+        clock::Clock as ClockTrait,
+    };
+
+    #[derive(Clone, Encode, Decode, PartialEq, Eq, Debug, TypeInfo)]
+    pub struct DistributionRecord<BlockNumber, Balance> {
+        pub block_number: BlockNumber,
+        pub attester_rewards: Balance,
+        pub collator_rewards: Balance,
+        pub executor_rewards: Balance,
+        pub treasury_rewards: Balance,
+        pub available: Balance,
+        pub distributed: Balance,
+    }
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: Currency<Self::AccountId>;
+
+        type TreasuryAccount: Get<Self::AccountId>;
+        /// The total inflation per year, expressed as a Perbill.
+        ///
+        /// Default: 4.4% (44_000_000 / 1_000_000_000)
+        #[pallet::constant]
+        type TotalInflation: Get<Perbill>;
+
+        /// The attester's portion of the total inflation, expressed as a Perbill.
+        ///
+        /// Default: 1.1% (11 / 100)
+        #[pallet::constant]
+        type AttesterInflation: Get<Perbill>;
+
+        /// The executor's portion of the total inflation, expressed as a Perbill.
+        ///
+        /// Default: 0.8% (8 / 100)
+        #[pallet::constant]
+        type ExecutorInflation: Get<Perbill>;
+
+        /// The collator's portion of the total inflation, expressed as a Perbill.
+        ///
+        /// Default: 0.5% (5 / 100)
+        #[pallet::constant]
+        type CollatorInflation: Get<Perbill>;
+
+        /// The treasury's portion of the total inflation, expressed as a Perbill.
+        ///
+        /// Default: 2% (20 / 100)
+        #[pallet::constant]
+        type TreasuryInflation: Get<Perbill>;
+
+        /// The number of blocks in one year.
+        ///
+        /// Default: 5_256_000 (assuming 6s block time)
```

Comment:
We set up t3rn with 12 secs iirc

Reply:
will update the comments, the runtime config calculates for 12s:

>     pub const OneYear: BlockNumber = 2_628_000; // (365.25 * 24 * 60 * 60) / 12; assuming 12s block time
>     pub const InflationDistributionPeriod: BlockNumber = 100_800; // (14 * 24 * 60 * 60) / 12; assuming one distribution per two weeks
>     pub const TreasuryAccount: AccountId = AccountId::new([51u8; 32]);
>     pub const AvailableBootstrapSpenditure: Balance = 1_000_000 * (TRN as Balance); // 1 MLN UNIT